kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mwp/MwpBannerProvider.java

```diff
@@ -0,0 +1,195 @@
+    }
+  
+    return null;
+  }
+
+  @Override
+  public List<Attachment> listOrganizationBannerImages(OrganizationId organizationId, BannerId bannerId) {
+    fi.otavanopisto.mwp.client.model.Banner mwpBanner = findBannerByBannerId(bannerId);
+    if (mwpBanner != null) {
+      Integer featuredMediaId = mwpBanner.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(featuredMediaId);
+        if ((featuredMedia != null) && (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE)) {
+          return Collections.singletonList(translateAttachment(featuredMedia));
+        }
+      }
+    }
+  
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Attachment findBannerImage(OrganizationId organizationId, BannerId bannerId, AttachmentId attachmentId) {
+    fi.otavanopisto.mwp.client.model.Banner banner = findBannerByBannerId(bannerId);
+    if (banner != null) {
+      Integer featuredMediaId = banner.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        AttachmentId mwpAttachmentId = getImageAttachmentId(featuredMediaId);
+        if (!idController.idsEqual(attachmentId, mwpAttachmentId)) {
+          return null;
+        }
+        
+        fi.otavanopisto.mwp.client.model.Attachment attachment = findMedia(featuredMediaId);
+        if (attachment != null) {
+          return translateAttachment(attachment);
+        }
+      }
+    }
+  
+    return null;
+  }
+
+  @Override
+  public AttachmentData getBannerImageData(OrganizationId organizationId, BannerId bannerId, AttachmentId attachmentId,
+      Integer size) {
+    
+    Integer mediaId = getMediaId(attachmentId);
+    if (mediaId == null) {
+      return null;
+    }
+    
+    fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(mediaId);
+    if (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE) {
+      AttachmentData imageData = getImageData(featuredMedia.getSourceUrl());
+      
+      if (size != null) {
+        return scaleImage(imageData, size);
+      } else {
+        return imageData;
+      }
+      
+    }
+    
+    return null;
+  }
+
+  private fi.otavanopisto.mwp.client.model.Banner findBannerByBannerId(BannerId bannerId) {
+    BannerId kuntaApiId = idController.translateBannerId(bannerId, MwpConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      logger.severe(String.format("Failed to convert %s into MWP id", bannerId.toString()));
+      return null;
+    }
+    
+    ApiResponse<fi.otavanopisto.mwp.client.model.Banner> response = mwpApi.getApi().wpV2BannerIdGet(kuntaApiId.getId(), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding banner failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private List<Banner> translateBanners(List<fi.otavanopisto.mwp.client.model.Banner> mwpBanners) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateBanners" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)